internal/handlers/common/projection.go

```diff
@@ -14,10 +14,47 @@
 
 package common
 
-import "github.com/FerretDB/FerretDB/internal/types"
+import (
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
 
 // ProjectDocuments modifies given documents in places according to the given projection.
 func ProjectDocuments(docs []*types.Document, projection *types.Document) error {
-	// TODO
+	if projection.Len() == 0 {
+		return nil
+	}
+	projectionMap := projection.Map()
+	for i := 0; i < len(docs); i++ {
+		for k := range docs[i].Map() {
+			if k == "_id" {
+				continue
+			}
+
+			v, ok := projectionMap[k]
+			if !ok {
+				docs[i].Remove(k)
```

Comment:
Do we have a dance test for that? It is not clear from documentation how MongoDB behaves there

Reply:
no, dance has no tests for it at all,
 can it be merged to not to stop Dmitry?

Reply:
Sure